File: challenging/14_sparse_matrix_multiplication/sparse_matrix_multiplication_test.py
from sparse_matrix_multiplication import sparse_matrix_multiplication
import unittest

class TestProgram(unittest.TestCase):
    def test_case_1(self):
        matrix_a = [
            [0, 2, 0],
            [0, -3, 5],
        ]
        matrix_b = [
            [0, 10, 0],
            [0, 0, 0],
            [0, 0, 4],
        ]
        expected = [
            [0, 0, 0],
            [0, 0, 20],
        ]
        actual = sparse_matrix_multiplication(matrix_a, matrix_b)
        self.assertEqual(actual, expected)

File: challenging/14_sparse_matrix_multiplication/sparse_matrix_multiplication.test.js
const sparse_matrix_multiplication = require('./sparse_matrix_multiplication');

describe("sparse matrix mult", () => {
  

  test('Test case 1', () => {
      const matrix_a = [
          [0, 2, 0],
          [0, -3, 5],
      ];
      const matrix_b = [
          [0, 10, 0],
          [0, 0, 0],
          [0, 0, 4],
      ];
      const expected = [
          [0, 0, 0],
          [0, 0, 20],
      ];
      const actual = sparse_matrix_multiplication(matrix_a, matrix_b);
      expect(actual).toEqual(expected);
  });
});
